x/stake/endblock.go

```diff
@@ -24,6 +34,7 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) (validatorUpdates []abci.Valid
 			panic("sideChainIds does not equal to sideChainStores")
 		}
 	}
+	ctx.EventManager().EmitEvents(events)
```

Comment:
Let us have a test that whether event is part of consensus, if not we can simplify the logic and reuse code here.

Reply:
Done